vasyl_varyshchuk/Week_2/task_1/func_in_threads.py

```diff
@@ -0,0 +1,98 @@
+""""
+This module returns random size of file name from list by processing elements in separate threads
+"""
+
+import random
+import Queue
+from threading import Thread
+
+
+number_of_threads = 5
+len_of_list = number_of_threads * 100
+
+
+def recognize_file_type(file_name):
+    """This function returns extension of file passing file name as parameter"""
+    _, extension = file_name.split('.')
+    return extension
+
+
+def generate_file_names_list():
+    """This function generate list of files with random names. Length of list is specified by global variable
+    len_of_list which value depend on number on thread."""
+    names = ['na', 'ko', 'mi', 'fo', 'ad', 're', 'du', 'ly', 'ke', 'su', 'bo', 'na']
+    extensions = ['.txt', '.zip', '.ini', '.jpg', '.ttf']
+    file_names = []
+
+    for i in range(len_of_list):
+        file_name = ''
+        for j in range(3):
+            index = names.index(random.choice(names))
+            file_name += names[index]
+        ext = extensions.index(random.choice(extensions))
+        file_name += extensions[ext]
+        file_names.append(file_name)
+
+    return file_names
+
+
+def txt_handler(out_queue):
+    out_queue.put(random.randint(100, 450))
+
+
+def zip_handler(out_queue):
+    out_queue.put(random.randint(500, 700))
+
+
+def jpg_handler(out_queue):
+    out_queue.put(random.randint(800, 1000))
+
+
+def ttf_handler(out_queue):
+    out_queue.put(random.randint(1100, 1300))
+
+
+def ini_handler(out_queue):
+        out_queue.put(random.randint(1300, 1500))
+
+
+handlers = {
+    'txt': txt_handler,
+    'zip': zip_handler,
+    'jpg': jpg_handler,
+    'ttf': ttf_handler,
+    'ini': ini_handler
+ }
+
+
+def run_executors(list_of_file_names):
+    """Function which returns random size for file names from a list."""
+    for item in list_of_file_names:
+        threads = []
+        extension = recognize_file_type(item)
+
+        my_queue = Queue.Queue()
+
+        for i in range(0, number_of_threads):
```

Comment:
You don't need this extra zero. Range works in this way by default.

Reply:
Sure, I understand it. Using zero in range() is my bad habit. I'll try to omit it next time.